### ISSUE TITLE
I'm getting SegFaults when running compression

### ISSUE THREAD

**Issue by zacksiri** [ralscha/parcel-plugin-compress#5]
Not sure why this is happening it seems quite un-predictable. I'm getting segfaults during compression process.

**Comment by ralscha** [comment]
Hi. Can you provide some more information. Node.js version, operating system

**Comment by zacksiri** [comment]
@ralscha Yep, I'm running build on Netlify using `Ubuntu 16.04` which is using node `v10.15.3` and yarn `1.13.0` i believe it is running inside of docker. I also have the example code here 
https://www.github.com/artellectual/artellectual 

You might see it as 'success' for netlify build but the only way I got it working is keep requesting re-build and eventually the build succeeds.

edit: This also happens on my macOS running node 11 however it is less frequent.

additionally here is an example log

```
11:50:42 AM: Build ready to start
11:50:44 AM: build-image version: 324ec043422499a87b63cac1f1dabeefe6dca19d
11:50:44 AM: build-image tag: v3.0.2
11:50:44 AM: buildbot version: ef2e26260c41679f4cdeaebbf93370345c9fecf7
11:50:44 AM: Fetching cached dependencies
11:50:44 AM: Starting to download cache of 102.7MB
11:50:45 AM: Finished downloading cache in 751.565241ms
11:50:45 AM: Starting to extract cache
11:50:49 AM: Finished extracting cache in 4.008684271s
11:50:49 AM: Finished fetching cache in 4.851110745s
11:50:49 AM: Starting to prepare the repo for build
11:50:49 AM: Preparing Git Reference refs/heads/master
11:50:50 AM: Starting build script
11:50:50 AM: Installing dependencies
11:50:50 AM: Started restoring cached node version
11:50:52 AM: Finished restoring cached node version
11:50:53 AM: v10.15.3 is already installed.
11:50:54 AM: Now using node v10.15.3 (npm v6.4.1)
11:50:54 AM: Attempting ruby version 2.6.2, read from environment
11:50:56 AM: Using ruby version 2.6.2
11:50:56 AM: Using PHP version 5.6
11:50:56 AM: Started restoring cached node modules
11:50:56 AM: Finished restoring cached node modules
11:50:56 AM: Started restoring cached yarn cache
11:50:56 AM: Finished restoring cached yarn cache
11:50:57 AM: Installing NPM modules using Yarn version 1.13.0
11:50:58 AM: yarn install v1.13.0
11:50:58 AM: [1/4] Resolving packages...
11:50:59 AM: success Already up-to-date.
11:50:59 AM: Done in 0.81s.
11:50:59 AM: NPM modules installed using Yarn
11:50:59 AM: Started restoring cached go cache
11:50:59 AM: Finished restoring cached go cache
11:50:59 AM: unset GOOS;
11:50:59 AM: unset GOARCH;
11:50:59 AM: export GOROOT='/opt/buildhome/.gimme/versions/go1.12.linux.amd64';
11:50:59 AM: export PATH="/opt/buildhome/.gimme/versions/go1.12.linux.amd64/bin:${PATH}";
11:50:59 AM: go version >&2;
11:50:59 AM: export GIMME_ENV='/opt/buildhome/.gimme/env/go1.12.linux.amd64.env';
11:50:59 AM: go version go1.12 linux/amd64
11:50:59 AM: Installing missing commands
11:50:59 AM: Verify run directory
11:50:59 AM: Executing user command: yarn run build
11:51:00 AM: yarn run v1.13.0
11:51:00 AM: $ NODE_ENV=production parcel build app/index.html
11:51:44 AM: ✨  Built in 43.89s.
11:51:44 AM: dist/all.120bfc90.js.map                         ⚠️  1.25 MB      93ms
11:51:44 AM: dist/all.120bfc90.js                             ⚠️  1.07 MB    17.40s
11:51:44 AM: dist/app.d31b5a3f.js.map                          885.38 KB     130ms
11:51:44 AM: dist/app.d31b5a3f.js                              360.77 KB    42.32s
11:51:44 AM: dist/tailwind.dc8a1dd6.css                         334.6 KB    36.52s
11:51:44 AM: dist/tailwind.dc8a1dd6.css.map                     131.4 KB     110ms
11:51:44 AM: dist/animate.a91491eb.css.map                     101.75 KB      70ms
11:51:44 AM: dist/animate.a91491eb.css                          56.66 KB     6.34s
11:51:44 AM: dist/open-sans-latin-800.9b962ac8.woff             18.21 KB     4.84s
11:51:44 AM: dist/open-sans-latin-700.3c63fad8.woff             18.04 KB     4.84s
11:51:44 AM: dist/open-sans-latin-600.ce28ea55.woff             17.87 KB     4.74s
11:51:44 AM: dist/open-sans-latin-300.f229a5dd.woff             17.85 KB     4.69s
11:51:44 AM: dist/open-sans-latin-400.af8aa85c.woff             17.29 KB     4.69s
11:51:44 AM: dist/open-sans-latin-800italic.423084fc.woff          17 KB    10.38s
11:51:44 AM: dist/open-sans-latin-300italic.f7c1dfab.woff       16.89 KB     4.69s
11:51:44 AM: dist/open-sans-latin-600italic.196cdeb2.woff       16.72 KB     4.74s
11:51:44 AM: dist/open-sans-latin-400italic.bf1ae893.woff       16.67 KB     4.69s
11:51:44 AM: dist/open-sans-latin-700italic.29096f48.woff       16.67 KB     4.84s
11:51:44 AM: dist/open-sans-latin-800.c553d989.woff2            14.39 KB     4.84s
11:51:44 AM: dist/open-sans-latin-700.80a32167.woff2            14.38 KB     4.75s
11:51:44 AM: dist/open-sans-latin-300.0b5683b3.woff2            14.22 KB     4.68s
11:51:44 AM: dist/open-sans-latin-600.a85d2156.woff2             14.2 KB     4.74s
11:51:44 AM: dist/open-sans-latin-400.2c405b88.woff2            13.72 KB     4.69s
11:51:44 AM: dist/open-sans-latin-800italic.bcfd6a54.woff2      13.34 KB    10.38s
11:51:44 AM: dist/open-sans-latin-300italic.7f9463b8.woff2      13.21 KB     4.69s
11:51:44 AM: dist/open-sans-latin-600italic.7ac5c87b.woff2      13.21 KB     4.74s
11:51:44 AM: dist/open-sans-latin-700italic.ac7bbcb9.woff2      13.11 KB     4.84s
11:51:44 AM: dist/open-sans-latin-400italic.4582af1e.woff2       13.1 KB     4.67s
11:51:44 AM: dist/app.4fd93520.css.map                           5.08 KB     110ms
11:51:44 AM: dist/favicon.2c16c84d.png                           4.96 KB    11.22s
11:51:44 AM: dist/root.ee19146f.js.map                           4.69 KB       8ms
11:51:44 AM: dist/root.ee19146f.js                               4.49 KB     5.21s
11:51:44 AM: dist/app.4fd93520.css                               2.86 KB    38.86s
11:51:44 AM: dist/not_found.e0347a9f.js                           1.6 KB     743ms
11:51:44 AM: dist/codemy.96b6c531.js                             1.08 KB    36.53s
11:51:44 AM: dist/tailwind.dc8a1dd6.js                           1.08 KB    36.52s
11:51:44 AM: dist/logo.2b725cf3.svg                                805 B     8.01s
11:51:44 AM: dist/not_found.e0347a9f.js.map                        782 B     110ms
11:51:44 AM: dist/index.html                                       715 B     1.50s
11:51:44 AM: dist/root.c2861047.css.map                            642 B     110ms
11:51:44 AM: dist/root.c2861047.css                                469 B     4.38s
11:51:44 AM: dist/codemy.96b6c531.css.map                          297 B     110ms
11:51:44 AM: dist/codemy.96b6c531.css                              252 B    36.52s
11:51:44 AM: 🗜️  Compressing bundled files...
11:53:23 AM: ✨  Compressed in 98.73s.
11:53:23 AM: dist/all.120bfc90.js.map.gz                         410.47 KB    17.626s
11:53:23 AM: dist/all.120bfc90.js.gz                             379.06 KB    16.784s
11:53:23 AM: dist/all.120bfc90.js.map.br                         356.77 KB    17.629s
11:53:23 AM: dist/all.120bfc90.js.br                             330.36 KB    16.781s
11:53:23 AM: dist/app.d31b5a3f.js.map.gz                         189.62 KB    20.281s
11:53:23 AM: dist/app.d31b5a3f.js.map.br                         169.88 KB    22.194s
11:53:23 AM: dist/app.d31b5a3f.js.gz                              88.56 KB     5.018s
11:53:23 AM: dist/app.d31b5a3f.js.br                              78.04 KB     6.032s
11:53:23 AM: dist/tailwind.dc8a1dd6.css.gz                        47.28 KB     6.023s
11:53:23 AM: dist/open-sans-latin-800.9b962ac8.woff.gz            18.12 KB     0.626s
11:53:23 AM: dist/open-sans-latin-800.9b962ac8.woff.br            18.08 KB     0.523s
11:53:23 AM: dist/open-sans-latin-700.3c63fad8.woff.gz            17.96 KB     0.792s
11:53:23 AM: dist/open-sans-latin-700.3c63fad8.woff.br            17.91 KB     0.791s
11:53:23 AM: dist/open-sans-latin-600.ce28ea55.woff.gz            17.78 KB     0.714s
11:53:23 AM: dist/open-sans-latin-300.f229a5dd.woff.gz            17.78 KB     0.321s
11:53:23 AM: dist/open-sans-latin-600.ce28ea55.woff.br            17.74 KB     0.714s
11:53:23 AM: dist/open-sans-latin-300.f229a5dd.woff.br            17.71 KB     0.263s
11:53:23 AM: dist/open-sans-latin-400.af8aa85c.woff.gz            17.22 KB     0.255s
11:53:23 AM: dist/open-sans-latin-400.af8aa85c.woff.br            17.15 KB     0.254s
11:53:23 AM: dist/tailwind.dc8a1dd6.css.br                        17.14 KB     6.022s
11:53:23 AM: dist/open-sans-latin-800italic.423084fc.woff.gz      16.91 KB     0.714s
11:53:23 AM: dist/open-sans-latin-800italic.423084fc.woff.br      16.86 KB     0.677s
11:53:23 AM: dist/open-sans-latin-300italic.f7c1dfab.woff.gz      16.82 KB     0.239s
11:53:23 AM: dist/open-sans-latin-300italic.f7c1dfab.woff.br      16.75 KB     0.238s
11:53:23 AM: dist/open-sans-latin-600italic.196cdeb2.woff.gz      16.62 KB     0.498s
11:53:23 AM: dist/open-sans-latin-600italic.196cdeb2.woff.br      16.58 KB     0.499s
11:53:23 AM: dist/open-sans-latin-700italic.29096f48.woff.gz      16.58 KB     0.679s
11:53:23 AM: dist/open-sans-latin-400italic.bf1ae893.woff.gz      16.58 KB     0.705s
11:53:23 AM: dist/open-sans-latin-700italic.29096f48.woff.br      16.54 KB     0.679s
11:53:23 AM: dist/open-sans-latin-400italic.bf1ae893.woff.br      16.53 KB     0.705s
11:53:23 AM: dist/open-sans-latin-800.c553d989.woff2.gz           14.39 KB     0.245s
11:53:23 AM: dist/open-sans-latin-700.80a32167.woff2.br           14.37 KB     0.258s
11:53:23 AM: dist/open-sans-latin-700.80a32167.woff2.gz           14.37 KB     0.258s
11:53:23 AM: dist/open-sans-latin-300.0b5683b3.woff2.gz           14.22 KB     0.251s
11:53:23 AM: dist/open-sans-latin-600.a85d2156.woff2.gz            14.2 KB     0.274s
11:53:23 AM: dist/open-sans-latin-400.2c405b88.woff2.gz           13.72 KB     0.272s
11:53:23 AM: dist/open-sans-latin-300italic.7f9463b8.woff2.gz     13.21 KB     0.275s
11:53:23 AM: dist/open-sans-latin-600italic.7ac5c87b.woff2.gz     13.21 KB      0.23s
11:53:23 AM: dist/open-sans-latin-700italic.ac7bbcb9.woff2.gz     13.11 KB     0.242s
11:53:23 AM: dist/open-sans-latin-400italic.4582af1e.woff2.gz      13.1 KB     0.264s
11:53:23 AM: dist/animate.a91491eb.css.map.gz                      6.19 KB     4.801s
11:53:23 AM: dist/animate.a91491eb.css.map.br                       5.5 KB     4.801s
11:53:23 AM: dist/favicon.2c16c84d.png.br                          4.84 KB     0.144s
11:53:23 AM: dist/favicon.2c16c84d.png.gz                          4.83 KB     0.147s
11:53:23 AM: dist/tailwind.dc8a1dd6.css.map.gz                     4.15 KB     11.15s
11:53:23 AM: dist/animate.a91491eb.css.gz                          3.81 KB     1.501s
11:53:23 AM: dist/animate.a91491eb.css.br                          3.44 KB     1.501s
11:53:23 AM: dist/tailwind.dc8a1dd6.css.map.br                     3.19 KB     11.15s
11:53:23 AM: dist/root.ee19146f.js.gz                              1.48 KB     0.094s
11:53:23 AM: dist/root.ee19146f.js.br                              1.34 KB     0.094s
11:53:23 AM: dist/root.ee19146f.js.map.gz                          1.29 KB      0.66s
11:53:23 AM: dist/root.ee19146f.js.map.br                          1.17 KB     0.661s
11:53:23 AM: dist/not_found.e0347a9f.js.gz                           798 B      0.42s
11:53:23 AM: dist/app.4fd93520.css.map.gz                            768 B      0.81s
11:53:23 AM: dist/not_found.e0347a9f.js.br                           718 B      0.42s
11:53:23 AM: dist/app.4fd93520.css.map.br                            660 B     0.252s
11:53:23 AM: dist/app.4fd93520.css.gz                                632 B     2.933s
11:53:23 AM: dist/codemy.96b6c531.js.gz                              551 B      0.26s
11:53:23 AM: dist/tailwind.dc8a1dd6.js.gz                            551 B     0.284s
11:53:23 AM: dist/app.4fd93520.css.br                                524 B      0.81s
11:53:23 AM: dist/tailwind.dc8a1dd6.js.br                            497 B     0.284s
11:53:23 AM: dist/codemy.96b6c531.js.br                              490 B      0.26s
11:53:23 AM: dist/logo.2b725cf3.svg.gz                               417 B     0.332s
11:53:23 AM: dist/root.c2861047.css.map.gz                           403 B     0.766s
11:53:23 AM: dist/logo.2b725cf3.svg.br                               387 B     0.333s
11:53:23 AM: dist/not_found.e0347a9f.js.map.gz                       360 B     0.144s
11:53:23 AM: dist/root.c2861047.css.map.br                           341 B     0.766s
11:53:23 AM: dist/index.html.gz                                      330 B     0.387s
11:53:23 AM: dist/not_found.e0347a9f.js.map.br                       321 B     0.144s
11:53:23 AM: dist/root.c2861047.css.gz                               290 B      0.14s
11:53:23 AM: dist/index.html.br                                      231 B     0.387s
11:53:23 AM: dist/root.c2861047.css.br                               230 B      0.14s
11:53:23 AM: dist/codemy.96b6c531.css.map.gz                         225 B     0.089s
11:53:23 AM: dist/codemy.96b6c531.css.map.br                         214 B     0.089s
11:53:23 AM: dist/codemy.96b6c531.css.gz                             211 B     0.092s
11:53:23 AM: dist/codemy.96b6c531.css.br                             183 B     0.091s
11:53:25 AM: Segmentation fault (core dumped)
11:53:25 AM: error Command failed with exit code 139.
11:53:25 AM: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
11:53:25 AM: Caching artifacts
11:53:25 AM: Started saving node modules
11:53:25 AM: Finished saving node modules
11:53:25 AM: Started saving yarn cache
11:53:25 AM: Finished saving yarn cache
11:53:25 AM: Started saving pip cache
11:53:25 AM: Finished saving pip cache
11:53:25 AM: Started saving emacs cask dependencies
11:53:25 AM: Finished saving emacs cask dependencies
11:53:25 AM: Started saving maven dependencies
11:53:25 AM: Finished saving maven dependencies
11:53:25 AM: Started saving boot dependencies
11:53:25 AM: Finished saving boot dependencies
11:53:25 AM: Started saving go dependencies
11:53:25 AM: Finished saving go dependencies
11:53:25 AM: Error running command: Build script returned non-zero exit code: 1
11:53:25 AM: Failing build: Failed to build site
11:53:25 AM: failed during stage 'building site': Build script returned non-zero exit code: 1
11:53:25 AM: Finished processing build request in 2m41.482275361s
11:53:25 AM: Shutting down logging, 0 messages pending
```

**Comment by zacksiri** [comment]
So after doing a bit more investigation I was able to re-produce this problem on Alpine Linux 3.9 running node 11.3.0 as well. It's funny because the compression actually completes successfully for all the files. Then I get a Segfault

**Comment by zacksiri** [comment]
@ralscha so i absolutely could not get builds to work with node 10, 11 works sometimes, doesn't work sometimes. I downgraded to node 8 and it worked straight away everytime.

**Comment by ralscha** [comment]
I released a new version 1.0.8 that uses node-zopfli-es (for issue #6)
Maybe you can give this version a try. 

Can you also enable gzip and/or brotli, to see which compression causes the problem

`package.json`
```
"compress": {
   "gzip": {   
      "enabled": false
   },
   "brotli": {
       "enabled": false
   }
}
```

**Comment by zacksiri** [comment]
@ralscha I tried it out and it solves the problem on Node 10 / 11 for now all seems to be fine.
